core/src/main/java/org/wso2/carbon/kernel/utils/Utils.java

```diff
@@ -156,4 +173,207 @@ public static void checkSecurity() {
+     * @param tagName Tag name of the element
+     * @param text    Text of the element
+     * @return XML element in String format
+     */
+    public static String createXmlElement(String tagName, String text) {
+        return "<" + tagName + ">" + text + "</" + tagName + ">";
+    }
+
+    /**
+     * This method converts a given XML String to JSON format
+     *
+     * @param xmlString XML String that needs to be converted to JSON format
+     * @return String in JSON format
+     */
+    public static String convertXMLToJSON(String xmlString) {
+        String jsonString;
+        try {
+            JSONObject xmlJSONObj = XML.toJSONObject(xmlString);
+            jsonString = xmlJSONObj.toString();
+        } catch (JSONException e) {
+            logger.error("Exception occurred while converting XML to JSON: ", e);
+            throw new RuntimeException("Exception occurred while converting XML to JSON: ", e);
+        }
+        return jsonString;
+    }
+
+    /**
+     * This method converts a Properties file to XML formatted String
+     *
+     * @param inputStream InputStream of the Properties file
+     * @return String in XML format
+     */
+    public static String convertPropertiesToXml(InputStream inputStream, String rootElement) {
+        StringBuilder stringBuilder;
+        java.util.Properties properties = new java.util.Properties();
+        try {
+            properties.load(inputStream);
+            stringBuilder = new StringBuilder();
+            properties.entrySet().forEach(entry -> stringBuilder
+                    .append(createXmlElement(entry.getKey().toString(), entry.getValue().toString())));
+
+        } catch (IOException e) {
+            logger.error("Exception occurred while converting Properties to XML: ", e);
+            throw new RuntimeException("Exception occurred while converting Properties to XML: ", e);
+        }
+        String xmlString = createXmlElement(rootElement, stringBuilder.toString());
+        //Need to add a root element
+        return prettyFormatXMLString(xmlString);
+    }
+
+    /**
+     * This method converts a given XML String to Properties format
+     *
+     * @param xmlString XML String that needs to be converted to Properties format
+     * @return String in Properties format
+     */
+    public static String convertXMLToProperties(String xmlString, String rootElement) {
+        String jsonString = convertXMLToJSON(xmlString);
+        Yaml yaml = new Yaml();
+        Map map = yaml.loadAs(jsonString, Map.class);
+        //Remove root element
+        if (map.containsKey(rootElement)) {
+            map = (Map) map.get(rootElement);
+        } else {
+            throw new RuntimeException("Root element not found when converting XML to Properties");
+        }
+        StringBuilder stringBuilder = new StringBuilder();
+        String tempString;
+        for (Object entryObject : map.entrySet()) {
+            Map.Entry entry = (Map.Entry) entryObject;
```

Comment:
This is fixed in the following commit.

https://github.com/jsdjayanga/carbon-kernel/blob/84257f8c953e5c174ff21f3febb70797228a9d8b/core/src/main/java/org/wso2/carbon/kernel/utils/Utils.java#L324